gprofiler/profiler_state.py

```diff
@@ -0,0 +1,13 @@
+from threading import Event
+
+
+class ProfilerState:
+    def __init__(
+        self,
+        stop_event: Event,
+        storage_dir: str,
+        profile_spawned_processes: bool,
```

Comment:
Other fields I think can be added here (because they are generic for all) -
* `insert_dso_name`
* `profiling_mode`